### DISCUSSION TITLE
Cloud Storage Integration (Dropbox, Box, Drive, Onedrive, Mega, etc)

### DISCUSSION THREAD

**Discussion by kevinhammons** [PaulWoitaschek/Voice#1294] [Ideas]
Feature request : Being able to stream audiobooks stored on cloud accounts would be awesome.  Astro file manager does it rather well for file management and Beat does a good job for music.

**Comment by PaulWoitaschek** [comment]
This request has been here seeral times but is really low priority for me right now. But if someone wants to work on it, I'll be happy to merge it.

**Comment by kolonuk** [comment]
Just like to register my interest in this too.  Thinking about getting in to Android development, so might look into this at some point... No promises though!

**Comment by Domoel** [comment]
+1

**Comment by agabrielluz** [comment]
Hi, I love your app. S2. There's an app for Windows Phone 8.1 called Audiobooked, they've add Dropbox integration and it works like a charm. You click on Browse Dropbox and then your folders are shown. Cover art support for Dropbox would be good. For example, on the Dropbox folder, we create a JPG cover file with .MAPC extension with the same exactly filename as the audiobook so that Material Audiobook Player can show the Cloud files with Cover! It would be cool. Example:
audiobook.m4b
audiobook.mapc

Note: Material Audiobook Player would hide the .mapc file and show the m4b file with the cover instead. In the future, .MAPC file could be used as a metadata file so that it loads metada faster instead of reading the entire m4b. Useful for Cloud shares.

**Comment by PaulWoitaschek** [comment]
Kodi integration can be done once this is coming.

**Comment by PaulWoitaschek** [comment]
Goes along with #295

**Comment by marius** [comment]
It may go along with #295 but it is a different thing. Imo this bug should be opened again.

**Comment by kevmitch** [comment]
I believe that @PaulWoitaschek's intention is that closing this issue doesn't mean that it won't be solved, but that solving #295 will also solve this one. So from the perspective of the developer, there is no need to identify two distinct things on the to-do list.

**Comment by p3ngwin** [comment]
**+1 for cloud storage support,** in my case i find it strange this Android Application has no Google Drive support.

All my Audiobooks are in my Google Drive folder, so this app was completely unusable for me :(

I hope to come back to it in future if support is added :)

**Comment by eturk1** [comment]
+1 for cloud storage support
any movement on this?

**Comment by versavius** [comment]
With these cloud accounts, are you syncing the items to your device then playing from your device or directly streaming from the cloud.

So Dropbox and Google Drive seem it is possible:

https://lifehacker.com/how-to-turn-google-drive-into-your-personal-media-serve-1551083632

Whats the advantage of playing it over the cloud rather than from your device directly ? I prefer to keep audio on the device due to drops in mobile connection (3G/4G) etc

**Comment by p3ngwin** [comment]
My reason for wanting cloud, in particular Google Drive support, is because that's where i store all my Audio books, so i can access them on whatever screen i'm using.

I just want to be able to use my mobile android devices to tap into my existing resources, I.E. GDrive and my Audio Books, so i don't have to make duplicate files and folders, etc manually micro-managing everything.

I already _have_ content, i just want an audio book player to access it quick, easy, and effortlessly, so i can enjoy my books :)

**Reply by im-umairaslami** [reply]
Hey! I totally get the need for easy access to your audiobooks across multiple devices. If you’re looking for an alternative to Google Drive with even more flexibility, you might want to check out TeraBox. With TeraBox, you can store and manage your audio content effortlessly in the cloud, and access it from any device. Plus, there's a modded APK version that makes it easier to handle large media files without restrictions. <a href="https://teraboxmod.com">Terabox Mod APK</a>
 for more details and to get started! It’s an excellent solution if you’re looking to streamline your audiobook library and access everything quickly and hassle-free.

**Comment by gsarpy** [comment]
+1 I'd love to see this. It would make it really easy to add books to my dropbox via my computer and seamlessly access them on the phone. I'd create a PR for this but I am not a Java dev so I doubt it'd be good code.

**Comment by barrycraig** [comment]
+1 from me too. The ability to browse a folder on Google drive and click 'download' on the book I want to listen to next would streamline what is currently a clunky procedure of downloading via a third party app and moving the folder manually.

**Comment by os97673** [comment]
#884, #855, and #665 about the same, perhaps it make sense to close them since they do not add anything and keep this issue to track/discuss this feature.

**Comment by jtagcat** [comment]
You can make files available offline, point Voice to the files found on the filesystem. Or self-host Nextcloud with your cloud storage as a backend, sync Nextcloud to your device.
More advanced workaround would be to use cron+rclone.

**Comment by Stephan-4711** [comment]
Plus 10 me and my friends are waiting too

**Comment by thw0rted** [comment]
I think Android added some kind of native support for this at some point.  My phone run Android 13 and in the stock Files app, I can see both Google Drive and Onedrive in the left sidebar.  In another thread, @PaulWoitaschek pointed out that you can actually pick those in the "add file" (i.e. single book) dialog.  It's a little tedious because I have to add each book individually, rather than a folder full of books, but other than that it has worked out great for me.

(If your books aren't in a single file yet, I highly recommend it especially for cloud storage, it makes management a lot easier.  I converted mine to M4A with https://www.freac.org/ .)

**Reply by Stephan-4711** [reply]
I just tried. It seems the file then is downloaded.
That's not really using the Cloud function.  A folder based usage and load it on demand would be much more satisfying

**Reply by thw0rted** [reply]
Oh that's the exact opposite of what I was looking for -- if I wanted to stream my audiobooks I'd just use the Audible cloud player.  I wanted a way to keep my whole collection in the cloud then "cache" the couple I'm actively listening to so I can play them offline.  It does mean periodically removing old ones and adding new ones in Voice, but I can always see what's available and it's easy to add a new one.  It's not like music where you can go through 20 files in an hour, so the extra overhead is manageable for me.

Streaming seems like it could be more difficult technically -- do the cloud storage providers even let you request the "middle" of a big file? -- but you can always ask.

**Comment by zoneh5** [comment]
[Jenny Mod Download PC ](https://getjennyminecraft.com/jenny-mod-mcpe/)requires installing Minecraft Java Edition and Forge to run the mod smoothly. After setting up Forge, download the Jenny Mod from a trusted source and place it in the mods folder. Launch Minecraft using the Forge profile, and the mod should be ready to use, adding new interactive features to the game.

**Reply by truh** [reply]
Wrong thread?

**Reply by thw0rted** [reply]
The problem with spam on GitHub is that only project maintainers can flag it :(
